### ISSUE TITLE
Do some research for event calendar and let us know your thoughts in comments

### ISSUE THREAD

**Issue by bjupreti** [bjupreti/karyakram-frontend#2]

**Comment by NavGithub15** [comment]
Upon my research I came across two libraries called React Big Calendar and FullCalendar stand out in terms of customization, features, and logic implementation. I also found a youtube video in which the person built a google calendar clone from scratch and his code is available on github and we can use his code which we can improvise and customize for our use case. Here is link to repo https://github.com/3stbn/google-calendar-clone and link to deployed site https://google-calendar-clone.netlify.app/

- React Big Calendar provides flexibility and customization, also it comes with no built-in styling. It supports selectable time-slots, drag-and-drop events, and various views. Extending functionality might have limits.

- FullCalendar offers a robust feature set, extensive APIs, and customizable views. It has event dragging and dropping, resizing, and selectable time-slots. It supports custom views and plugins for extensibility. Some premium features require purchase.

In conclusion, both libraries offer solid customization options and features. The choice depends on our specific needs and the level of customization we require.

**Comment by bjupreti** [comment]
Google calendar clone is pretty impressive, especially on desktop view. One thing to note here is it's not responsive and may add lots of work for us if we want to make our website responsive.

Few things I noted about Fullcalendar:
- It has [TS support](https://fullcalendar.io/docs/typescript) but we will have to setup our own compiler (for e.g., tsc, webpack, etc.)
- UI looks good
- [Example Project in TypeScript](https://github.com/fullcalendar/fullcalendar-examples/tree/main/typescript)
- [Example Project in React](https://github.com/fullcalendar/fullcalendar-examples/tree/main/react)

It looks like React big calendar also supports Typescript.

One difference I found is React big calendar uses [react@16](https://github.com/jquense/react-big-calendar/blob/master/package.json) whereas Fullcalendar uses [react@18](https://github.com/fullcalendar/fullcalendar-examples/blob/main/react/package.json). I'm leaning towards Fullcalendar. If you are cool, I say we go with FullCalendar with TypeScript

**Comment by bjupreti** [comment]
Decided to use Fullcalendar.
